common/network-common/src/test/java/org/apache/spark/network/TransportClientFactorySuite.java

```diff
@@ -117,7 +117,7 @@ private void testClientReuse(int maxConnections, boolean concurrent)
       }
 
       Assert.assertEquals(0, failed.get());
-      Assert.assertEquals(clients.size(), maxConnections);
+      Assert.assertTrue(clients.size() <= maxConnections);
```

Comment:
I suppose there's no way to ask it to not return a cached instance?
This seems to be testing that they're all different, but, that doesn't seem to be the current behavior, nor is it necessarily desirable. So, OK.